lib/genpack.js

```diff
@@ -0,0 +1,66 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const { encode, decode, ExtensionCodec } = require('@msgpack/msgpack')
+const emojiKeywords = require('emojilib')
+const emojiData = require('unicode-emoji-json')
+const orderedEmoji = require('unicode-emoji-json/data-ordered-emoji')
+const emojiComponents = require('unicode-emoji-json/data-emoji-components')
+
+const keywordsMap = new Map()
+const emmojiInfo = new Map()
+
+const emojiSymbols = Object.keys(emojiKeywords)
+for (const emojiSymbol of emojiSymbols) {
+  emojiKeywords[emojiSymbol].forEach(keyword => {
+    if (keywordsMap.has(keyword) === false) {
+      keywordsMap.set(keyword, [])
+    }
+    keywordsMap.get(keyword).push(emojiSymbol)
+  })
+  emmojiInfo.set(emojiSymbol, {
+    ...emojiData[emojiSymbol],
+    keywords: emojiKeywords[emojiSymbol]
+  })
+}
+
+const extensionCodec = new ExtensionCodec()
+const MAP_EXT_TYPE = 1 // Any in 0-127
+extensionCodec.register({
+  type: MAP_EXT_TYPE,
+  encode: (object) => {
+    if (object instanceof Map) {
+      return encode([...object])
+    } else {
+      return null
+    }
+  },
+  decode: (data) => {
+    const array = decode(data)
+    return new Map(array)
+  }
+})
+
+const packedEmoji = {
+  keywords: keywordsMap,
+  searchTerms: Array.from(keywordsMap.keys()),
+  emoji: emmojiInfo,
```

Comment:
I'm thinking we can pre-build all of the "alfredItems" and store them here. The benefit would be that we don't need to generate item data during search. The down side would be that we have to load an even larger data object on every search term. Alfred doesn't keep the script in memory. It's re-invoking it for every search term (i.e. every key press).